skimage/util/_flood_fill.py

```diff
@@ -0,0 +1,182 @@
+"""flood_fill.py - inplace flood fill algorithm
+
+This module provides a function to fill all equal (or within tolerance) values
+connected to a given seed point with a different value.
+"""
+
+import numpy as np
+
+
+def flood_fill(image, seed_point, new_value, selem=None, connectivity=None,
```

Comment:
Before diving into the code, I'd like to propose some changes to the API.
For me, one of the most common use cases for `flood fill` is to color the image regions given their boundaries. For this purpose, I'd expect the parameter `seed_point` to accept also a list of points with the corresponding labels, or a ndarray with binary (`True` marks the seed points of the same object class) or uint mask (non-zero elements mark the seed points and their values indicate the class index). I understand, that the current implementation is just one loop away from this API, but it is the change, I think, worth making.
Another thing: could you, please, explicitly specify the supported dimensionalities of `image`?

Reply:
There are not any restrictions on the rank of the input image.

I have a few reservations to complicating the API further.  First, our hypothetical user who wants to wield the tool multiple times can do so as easily as building that input list with multiple calls and the inplace kwarg.  Second, the output logic would branch much further; off the top of my head it would be dangerous to allow input like this and also permit indices to be true for memory reasons.  Third, with tolerance available the result of successive fills is not commutative and this will result in user confusion.

If such functionality would be desired it would entail a simple wrapper for any specific use case, but making it general is not trivial - and the added logic would slow its use as a low level tool.  If we wanted to ship a multi_fill function in the future that would be fine, exporting all of these issues to a separate function and discussion.

Open to potentially be convinced otherwise, but for now I think that is out scope here.